src/main/java/com/facebook/openwifirrm/optimizers/tpc/RandomTxPowerInitializer.java

```diff
@@ -30,50 +30,63 @@ public class RandomTxPowerInitializer extends TPC {
 	/** The RRM algorithm ID. */
 	public static final String ALGORITHM_ID = "random";
 
-	/** The fixed tx power (dBm). */
-	private final int txPower;
+	/** The PRNG instance. */
+	private final Random rng;
 
-	/** Constructor (uses random tx power). */
-	public RandomTxPowerInitializer(
-		DataModel model, String zone, DeviceDataManager deviceDataManager
-	) {
-		this(model, zone, deviceDataManager, getRandomTxPower());
-	}
+	/** Whether to set a different value per AP or use a single value for all APs */
+	private final boolean setDifferentTxPowerPerAp;
 
-	/** Constructor. */
+	/**
+	 * Constructor (uses random tx power per AP and allows passing in a custom
+	 * Random class to allow seeding).
+	 */
 	public RandomTxPowerInitializer(
 		DataModel model,
 		String zone,
 		DeviceDataManager deviceDataManager,
-		int txPower
+		boolean setDifferentTxPowerPerAp,
+		Random rng
 	) {
 		super(model, zone, deviceDataManager);
-		this.txPower = txPower;
+		this.setDifferentTxPowerPerAp = setDifferentTxPowerPerAp;
+		this.rng = rng;
+	}
+
+	/** Constructor (uses random tx power per AP). */
+	public RandomTxPowerInitializer(
+		DataModel model, String zone, DeviceDataManager deviceDataManager, boolean setDifferentTxPowerPerAp
+	) {
+		this(model, zone, deviceDataManager, setDifferentTxPowerPerAp, new Random());
+	}
+
+	/** Constructor (uses random tx power). */
+	public RandomTxPowerInitializer(
+		DataModel model, String zone, DeviceDataManager deviceDataManager
+	) {
+		this(model, zone, deviceDataManager, false, new Random());
```

Comment:
nit: put all these constructors in order from least -> most parameters, and do proper constructor chaining (see comment above)